pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/util/ElasticTestContainer.java

```diff
@@ -0,0 +1,88 @@
+package org.broadinstitute.dsm.util;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.typesafe.config.Config;
+import lombok.extern.slf4j.Slf4j;
+import org.broadinstitute.ddp.constants.ConfigFile;
+import org.broadinstitute.ddp.util.ConfigManager;
+import org.broadinstitute.ddp.util.ConfigUtil;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.statics.ApplicationConfigConstants;
+import org.testcontainers.elasticsearch.ElasticsearchContainer;
+
+@Slf4j
+public class ElasticTestContainer {
+
+    private static final String ELASTIC_IMAGE = "docker.elastic.co/elasticsearch/elasticsearch:7.17.10";
+    private static ElasticsearchContainer container;
+    private static boolean initialized = false;
+
+    public static synchronized void initialize() {
+        if (initialized) {
+            log.warn("ElasticTestContainer already initialized");
+            return;
+        }
+
+        ConfigManager configManager = ConfigManager.getInstance();
+        Config cfg = configManager.getConfig();
+        Boolean useDisposableTestDbs = ConfigUtil.getBoolIfPresent(cfg, ConfigFile.USE_DISPOSABLE_TEST_DB);
+        if (useDisposableTestDbs != null && !useDisposableTestDbs) {
+            log.info("Using external ElasticSearch instance (not using ElasticSearch test container)");
+            // do this here to avoid DSMConfig initialization issues
+            ElasticSearchUtil.initClient(cfg.getString(ConfigFile.ELASTICSEARCH_URL),
+                    cfg.getString(ConfigFile.ELASTICSEARCH_USERNAME),
+                    cfg.getString(ConfigFile.ELASTICSEARCH_PASSWORD), null);
+            initialized = true;
+            return;
+        }
+        log.info("Using ElasticSearch test container");
+
+        container = new ElasticsearchContainer(ELASTIC_IMAGE)
+                .withExposedPorts(9200)
+                .withEnv("xpack.security.enabled", "false")
```

Comment:
Can you drop a comment here and in the circle yaml so everyone knows that this is a deliberate choice *just for testing*?